src/main/code.ts

```diff
@@ -5,125 +5,422 @@
 // import Papa from "papaparse";
 // documentation: https://www.npmjs.com/package/papaparse
 
-if (figma.currentPage.selection.length != 1) {
-	figma.closePlugin("🛑 A single node must be selected");
-  } else if (figma.currentPage.selection[0].type !== 'INSTANCE') {
-	figma.closePlugin("🛑 Selected node must be an instance of a component");
-  } else {
-	// This shows the HTML page in "ui.html".
-	figma.showUI(__html__, { width: 300, height: 366, title: "Table Builder" });
-  }
-  
-  figma.ui.onmessage = msg => {
-	console.log(msg);
-	
-	if (msg.type === 'build-table') {
-	  
-	  var textNode = getFirstChildOfTypeText(figma.currentPage.selection[0])
-	  if (textNode == null) {
-		figma.closePlugin("🛑 Selected node must have a child of type TEXT");
-	  } else {
-		var fontName: FontName = textNode.fontName as FontName
-		Promise.all([
-		  figma.loadFontAsync({family: fontName['family'], style: fontName["style"]}),
-		]).then(() => { 
-			var headerCells = createTable(
-			  figma.currentPage.selection[0] as FrameNode, 
-			  msg.wrap_in_autolayout,
-			  msg.right_align_numbers,
-			  msg.data)
-			figma.currentPage.selection = headerCells;
+if (figma.command == "generate-table") {
+	if (figma.currentPage.selection.length != 1) {
+		figma.closePlugin("🛑 A single node must be selected");
+	} else if (figma.currentPage.selection[0].type !== 'INSTANCE') {
+		figma.closePlugin("🛑 Selected node must be an instance of a component");
+	} else {
+		// This shows the HTML page in "ui.html".
+		figma.showUI(__html__, { width: 300, height: 366, title: "Table Builder" });
+	}
+
+	figma.ui.onmessage = msg => {
+		console.log(msg);
+
+		if (msg.type === 'build-table') {
+
+			var textNode = getFirstChildOfTypeText(figma.currentPage.selection[0])
+			if (textNode == null) {
+				figma.closePlugin("🛑 Selected node must have a child of type TEXT");
+			} else {
+				var fontName: FontName = textNode.fontName as FontName
+				Promise.all([
+					figma.loadFontAsync({ family: fontName['family'], style: fontName["style"] }),
+				]).then(() => {
+					var headerCells = createTable(
+						figma.currentPage.selection[0] as FrameNode,
+						msg.wrap_in_autolayout,
+						msg.right_align_numbers,
+						msg.data)
+					figma.currentPage.selection = headerCells;
+					figma.closePlugin();
+				}
+				)
```

Comment:
Put closing brackets on the same line if corresponding opening brackets are also on the same line